pkg/lib/scoped/token_retriever.go

```diff
@@ -81,11 +81,11 @@ func getAPISecret(logger logrus.FieldLogger, kubeclient operatorclient.ClientInt
 // specific ServiceAccount via annotations kubernetes.io/service-account.name
 func filterSecretsBySAName(saName string, secrets *corev1.SecretList) map[string]*corev1.Secret {
 	secretMap := make(map[string]*corev1.Secret)
-	for _, ref := range secrets.Items {
+	for i, ref := range secrets.Items {
 		annotations := ref.GetAnnotations()
 		value := annotations[corev1.ServiceAccountNameKey]
 		if value == saName {
-			secretMap[ref.Name] = &ref
+			secretMap[ref.Name] = &secrets.Items[i]
 		}
 	}
```

Comment:
I created this [go playground code](https://go.dev/play/p/PosGriATixw) to showcase the issue.